Firebase.cpp

```diff
@@ -15,100 +15,241 @@
 //
 #include "Firebase.h"
 
-const char* firebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
-const uint16_t firebasePort = 443;
+namespace {
+const char* kFirebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
+const uint16_t kFirebasePort = 443;
 
-Firebase::Firebase(const String& host) : _host(host) {
-  _http.setReuse(true);
+String makeUrl(const String& path, const String& auth) {
+  String url;
+  if (path[0] != '/') {
+    url = "/";
+  }
+  url += path + ".json";
+  if (auth.length() > 0) {
+    url += "?auth=" + auth;
+  }
+  return url;
+}
+
+class FirebaseCall {
```

Comment:
Maybe we should rename that into `HTTPRequest` since there is not much specific to firebase in this helper (basically just the port and the fingerprint).